tests/FakeItEasy.Specs/CreationSpecs.cs

```diff
@@ -387,6 +388,19 @@ public void PublicDelegateWithPrivateTypeArgumentCannotBeFaked(Exception excepti
 "));
         }
 
+        [Scenario]
+        public void PublicDelegateWithAnonymousParameterCanBeFaked(IAmADelegateWithAnAnonymousParameter fake)
+        {
+            "Given a public delegate with an anonymous parameter "
+                .See<IAmADelegateWithAnAnonymousParameter>();
+
+            "When I create a fake of the delegate"
+                .x(() => fake = this.CreateFake<IAmADelegateWithAnAnonymousParameter>());
+
+            "Then the fake is created"
+                .x(() => fake.Should().BeAFake());
+        }

```

Comment:
Good catch! I wouldn't have thought of testing delegates

Reply:
Wasn't all me. Just asked VS "where do we get the name of a parameter" and followed the calls.